src/proxy/http/profiles.rs

```diff
@@ -0,0 +1,389 @@
+#![allow(dead_code)]
+
+extern crate tower_discover;
+
+use futures::Stream;
+use http;
+use indexmap::IndexMap;
+use regex::Regex;
+use std::iter::FromIterator;
+use std::sync::Arc;
+use std::{error, fmt};
+
+use transport::DnsNameAndPort;
+
+pub trait CanGetDestination {
+    fn get_destination(&self) -> Option<&DnsNameAndPort>;
+}
+
+pub type Routes = Vec<(RequestMatch, Route)>;
+
+pub trait GetRoutes {
+    type Stream: Stream<Item = Routes, Error = Error>;
+
+    fn get_routes(&self, dst: &DnsNameAndPort) -> Option<Self::Stream>;
+}
+
+pub trait WithRoute {
```

Comment:
Would love some doc comments on these traits explaining what they're for and how they're intended to be used...